utils/sampler/uniform_replacer.go

```diff
@@ -82,7 +76,7 @@ func (s *uniformReplacer) Next() (uint64, error) {
 		return 0, errOutOfRange
 	}
 
-	draw := uint64(s.rng.Int63n(int64(s.length-s.drawsCount))) + s.drawsCount
+	draw := s.rng.Uint64n(s.length-1-s.drawsCount) + s.drawsCount
```

Comment:
Note the `-1` here now because `Uint64n` returns values in the range `[0, n]` whereas `Int63n` returned values in the range `[0, n)`

Reply:
Would it be simpler to just have `Uint64n` return a number in `[0, n)`?

Reply:
Wondering if we should distinguish between these 2 random modes? Seems confusing that `*n` function now behaves differently (especially if it is different than the golang rand pkg: https://pkg.go.dev/math/rand#Int31n)?

Reply:
> Would it be simpler to just have Uint64n return a number in [0, n)?

This would mean that we would need to panic if `n == 0` which I was trying to avoid.

Reply:
Renamed `Uint64n` to `Uint64Inclusive` to be very clear